IronGram/src/main/java/com/theironyard/controllers/IronGramController.java

```diff
@@ -120,6 +145,38 @@ public Photo upload(HttpSession session, MultipartFile photo,
         }
 
         User user = userRepository.findFirstByName(username);
+
         return photoRepository.findByRecipientOrderByIdAsc(user);
     }
-}
+
+    @RequestMapping(path = "/public-photos", method = RequestMethod.GET)
+    public List<Photo> showPublicPhotos(HttpSession session){
+
+        String username = (String) session.getAttribute(SESSION_USERNAME);
+        User user = userRepository.findFirstByName(username);
+
+        if(username == null){
+            throw new NotLoggedIn();
```

Comment:
While this is totally my fault in class, this class should be named `NotLoggedInException`